client/landing/stepper/declarative-flow/site-setup-flow.ts

```diff
@@ -60,6 +61,16 @@ const siteSetupFlow: Flow = {
 				navigate( 'goals' );
 			}
 		}, [] );
+
+		const { setIntent, setGoals } = useDispatch( ONBOARD_STORE );
+		const { site } = useSiteData();
+
+		useEffect( () => {
+			if ( site?.options?.site_goals?.length ) {
```

Comment:
If a site has `site_goals` this would potentially override his selection in `goals` step if he uses the same site in the `site-setup` flow

Reply:
This is not true anymore right?

Reply:
yes with the last commit it's not 👍